src/main/java/com/fasterxml/jackson/dataformat/xml/XmlFactory.java

```diff
@@ -81,36 +81,44 @@
      * and this reuse only works within context of a single
      * factory instance.
      */
-    public XmlFactory() { this(null, null, null); }
+    public XmlFactory() { this(null, null, null, null); }
+
+    public XmlFactory(ClassLoader cl) {
+        this(null, null, null, cl);
+    }
 
     public XmlFactory(ObjectCodec oc) {
-        this(oc, null, null);
+        this(oc, null, null, null);
     }
 
     public XmlFactory(XMLInputFactory xmlIn) {
-        this(null, xmlIn, null);
+        this(null, xmlIn, null, null);
     }
     
     public XmlFactory(XMLInputFactory xmlIn, XMLOutputFactory xmlOut) {
-        this(null, xmlIn, xmlOut);
+        this(null, xmlIn, xmlOut, null);
     }
     
-    public XmlFactory(ObjectCodec oc, XMLInputFactory xmlIn, XMLOutputFactory xmlOut)
+    public XmlFactory(ObjectCodec oc, XMLInputFactory xmlIn, XMLOutputFactory xmlOut, ClassLoader cl)
     {
         this(oc, DEFAULT_XML_PARSER_FEATURE_FLAGS, DEFAULT_XML_GENERATOR_FEATURE_FLAGS,
-                xmlIn, xmlOut, null);
+                xmlIn, xmlOut, null, null);
```

Comment:
I think we should be plumbing through `cl` to the `protected` method rather than passing in `null`.